### PR TITLE
Document entity

### PR DESCRIPTION
**Pull request by apple502j**
~~Has more things to document, so draft for now.~~

Was originally planning to document `LivingEntity` etc, but this PR is very big right now and I don't want to make it bigger.

### CODE REVIEW

File: mappings/net/minecraft/entity/Entity.mapping [hunk 1]
```diff
@@ -1,4 +1,87 @@
...
+	COMMENT Entity ID is an integer used in networking, and is not saved on disk. UUID is
+	COMMENT used to identify an entity in NBT and other places where persistence is required.
+	COMMENT
+	COMMENT <h2 id="spawning">Creating and spawning entities</h2>
+	COMMENT Entities must be created first, which then can be added to a world ("spawning").
+	COMMENT There are multiple methods of doing this, shown on the table below:
+	COMMENT
+	COMMENT <div class="fabric">
+	COMMENT <table border=1>
+	COMMENT <caption>Creation & Spawning (at once)</caption>
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
	COMMENT <caption>Creation &amp; Spawning (at once)</caption>
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 2]
```diff
@@ -759,37 +1604,122 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 	METHOD method_5773 tick ()V
+		COMMENT Ticks this entity.
+		COMMENT
+		COMMENT @apiNote This can be overridden to add additional logics. {@code super.tick();}
+		COMMENT should be called in those cases.
+		COMMENT
+		COMMENT @implNote By default, this delegates all logics to {@link #baseTick}.
+		COMMENT
+		COMMENT @see LivingEntity#tickMovement
+		COMMENT @see MobEntity#mobTick
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see net.minecraft.entity.mob.MobEntity#mobTick
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 3]
```diff
@@ -509,20 +953,44 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 		COMMENT @see net.minecraft.entity.Entity#isTouchingWater()
 		COMMENT @see net.minecraft.entity.Entity#isBeingRainedOn()
 		COMMENT @see net.minecraft.entity.Entity#isInsideBubbleColumn()
+		COMMENT @see net.minecraft.entity.Entity#isInsideWaterOrBubbleColumn()
 	METHOD method_5638 shouldSetPositionOnLoad ()Z
 	METHOD method_5639 setOnFireFor (I)V
+		COMMENT Sets the entity on fire for {@code seconds} seconds.
+		COMMENT
+		COMMENT @implNote The actual duration can be reduced using the
+		COMMENT {@link ProtectionEnchantment$Type#FIRE fire protection} enchantment.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT {@linkplain net.minecraft.enchantment.ProtectionEnchantment.Type#FIRE
		COMMENT fire protection} enchantment.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 4]
```diff
@@ -509,20 +953,44 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT
+		COMMENT @see ProtectionEnchantment#transformFireDuration
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see net.minecraft.enchantment.ProtectionEnchantment#transformFireDuration
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 5]
```diff
@@ -413,29 +744,66 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT if the owner is a player; if the owner is a non-player entity, this returns
+		COMMENT the value of {@link net.minecraft.world.GameRules#DO_MOB_GRIEFING}, and ownerless
+		COMMENT projectiles are always allowed to modify the world.
+		COMMENT
+		COMMENT @see World#canPlayerModifyAt
 		ARG 1 world
 		ARG 2 pos
 	METHOD method_36974 tryCheckBlockCollision ()V
 	METHOD method_36975 playExtinguishSound ()V
+		COMMENT Plays the {@link SoundEvents#ENTITY_GENERIC_EXTINGUISH_FIRE} sound.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT Plays the {@link
		COMMENT net.minecraft.sound.SoundEvents#ENTITY_GENERIC_EXTINGUISH_FIRE} sound.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 6]
```diff
@@ -482,25 +885,66 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 	METHOD method_44329 getEventSource ()Lnet/minecraft/class_1297;
+		COMMENT {@return the entity to use as the game event source, or {@code null} if there is none}
+		COMMENT
+		COMMENT <p>By default this returns the entity itself; {@link ItemEntity} instead returns the
+		COMMENT thrower of the item entity.
 	METHOD method_5621 getMountedHeightOffset ()D
 	METHOD method_5622 onBlockCollision (Lnet/minecraft/class_2680;)V
+		COMMENT Called when this entity's collision box intersects {@code state}.
+		COMMENT
+		COMMENT @see AbstractBlock#onEntityCollision
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see net.minecraft.block.AbstractBlock#onEntityCollision
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 7]
```diff
@@ -482,25 +885,66 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 		ARG 1 state
 	METHOD method_5623 fall (DZLnet/minecraft/class_2680;Lnet/minecraft/class_2338;)V
+		COMMENT Called when the entity falls. Flying mobs should override this to do nothing.
+		COMMENT
+		COMMENT @implNote If on ground, this calls {@link Block#onLandedUpon}, which can add or
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @implNote If on ground, this calls {@link net.minecraft.block.Block#onLandedUpon}, which can add or
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 8]
```diff
@@ -707,44 +1453,143 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see #getLeashOffset
 	METHOD method_5752 getScoreboardTags ()Ljava/util/Set;
+		COMMENT {@return all scoreboard tags the entity belongs to}
+		COMMENT
+		COMMENT <p>Scoreboard tags are set using the {@linkplain net.minecraft.server.command.TagCommand
+		COMMENT /tag command}, and is different from entity type tags defined in data packs.
 	METHOD method_5753 isFireImmune ()Z
+		COMMENT {@return whether the entity is immune to {@linkplain DamageSource#setFire fire damage}}
+		COMMENT
+		COMMENT @see {@link EntityType.Builder#makeFireImmune}.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see EntityType.Builder#makeFireImmune
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 9]
```diff
@@ -681,9 +1406,20 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
 			COMMENT owner entity}
 		ARG 2 target
 	METHOD method_5724 updateVelocity (FLnet/minecraft/class_243;)V
+		COMMENT Updates the entity's velocity to add a vector in the direction of the entity's yaw
+		COMMENT whose absolute value is {@code movementInput} normalized and multiplied by {@code speed}.
+		COMMENT
+		COMMENT <p>This is usually called inside overridden {@link Entity#travel} if the entity is
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT <p>This is usually called inside overridden {@link LivingEntity#travel} if the entity is
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 10]
```diff
@@ -533,76 +1001,242 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 	METHOD method_5651 readNbt (Lnet/minecraft/class_2487;)V
 		ARG 1 nbt
 	METHOD method_5652 writeCustomDataToNbt (Lnet/minecraft/class_2487;)V
+		COMMENT Writes custom data to {@code nbt}. Subclasses has to implement this.
+		COMMENT
+		COMMENT <p>NBT is a storage format; therefore, a data from NBT is loaded to an entity instance's
+		COMMENT fields, which are used for other operations instead of the NBT. The data is written
+		COMMENT back to NBT when saving the entity.
+		COMMENT
+		COMMENT @see #readCustomDataToNbt
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #readCustomDataFromNbt
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 11]
```diff
@@ -611,47 +1245,125 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see World#canPlayerModifyAt
 		ARG 1 attacker
 	METHOD method_5699 dropStack (Lnet/minecraft/class_1799;F)Lnet/minecraft/class_1542;
+		COMMENT Drops {@code stack} at the entity's position with the given Y offset.
+		COMMENT
+		COMMENT @return the spawned item entity, or {@code null} if the stack is empty or if called
+		COMMENT on the client
+		COMMENT
+		COMMENT @see #dropItem(ItemConvertible)
+		COMMENT @see #dropItem(ItemConvertible, float)
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #dropItem(ItemConvertible, int)
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 12]
```diff
@@ -611,47 +1245,125 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see #hasPassenger
 		ARG 1 predicate
 	METHOD method_5705 getYaw (F)F
 		ARG 1 tickDelta
 	METHOD method_5706 dropItem (Lnet/minecraft/class_1935;)Lnet/minecraft/class_1542;
+		COMMENT Drops one {@code item} at the entity's position.
+		COMMENT
+		COMMENT @return the spawned item entity, or {@code null} if called on the client
+		COMMENT
+		COMMENT @see #dropItem(ItemConvertible, float)
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #dropItem(ItemConvertible, int)
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 13]
```diff
@@ -759,37 +1604,122 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 		ARG 5 fluidState
 		ARG 6 max
 	METHOD method_5775 dropStack (Lnet/minecraft/class_1799;)Lnet/minecraft/class_1542;
+		COMMENT Drops {@code stack} at the entity's position.
+		COMMENT
+		COMMENT @return the spawned item entity, or {@code null} if the stack is empty or if called
+		COMMENT on the client
+		COMMENT
+		COMMENT @see #dropItem(ItemConvertible)
+		COMMENT @see #dropItem(ItemConvertible, float)
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #dropItem(ItemConvertible, int)
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 14]
```diff
@@ -329,21 +522,45 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
 	METHOD method_30332 (Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Z
 		ARG 2 pos
 	METHOD method_30632 collidesWithStateAtPos (Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)Z
+		COMMENT {@return whether the entity collides with the block {@code state} at {@code pos}}
 		ARG 1 pos
 		ARG 2 state
 	METHOD method_30633 positionInPortal (Lnet/minecraft/class_2350$class_2351;Lnet/minecraft/class_5459$class_5460;)Lnet/minecraft/class_243;
+		COMMENT {@return the entity's position in the portal after teleportation}
+		COMMENT
+		COMMENT @see AreaHelper#entityPosInPortal
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see net.minecraft.world.dimension.AreaHelper#entityPosInPortal
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 15]
```diff
@@ -830,54 +1800,183 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see #streamPassengersAndSelf
+		COMMENT @see #getPassengersDeep
+		COMMENT @see #hasPlayerRider
 		ARG 1 passenger
 	METHOD method_5822 canUsePortals ()Z
+		COMMENT {@return whether the entity can use nether portals and end portals}
+		COMMENT
+		COMMENT <p>{@link net.minecraft.entity.boss.dragon.EnderDragonEntity},
+		COMMENT {@link net.minecraft.entity.boss.WitherEntity}, and {@link
+		COMMENT net.minecraft.entity.projectile.FishinBobberEntity} cannot use portals.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT net.minecraft.entity.projectile.FishingBobberEntity} cannot use portals.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 16]
```diff
@@ -193,13 +293,40 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT {@return whether the entity is crawling}
+		COMMENT
+		COMMENT <p>An entity is crawling if it is in swimming pose, but is not touching water.
+		COMMENT Players start crawling if they would otherwise collide with blocks (for example,
+		COMMENT when the player is in a 1 block tall tunnel).
+		COMMENT
+		COMMENT @see #isInSwimmingPose
 	METHOD method_20620 teleport (DDD)V
+		COMMENT Teleports the entity to the given position, loading the chunk with
+		COMMENT {@link ChunkTicketType#POST_TELEPORT}.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT {@link net.minecraft.server.world.ChunkTicketType#POST_TELEPORT}.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 17]
```diff
@@ -444,22 +812,57 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT if (this.world instanceof ServerWorld serverWorld) {
+		COMMENT   callback.accept(this.handler, serverWorld);
+		COMMENT }
+		COMMENT }</pre>
 		ARG 1 callback
 	METHOD method_42148 hasPrimaryPassenger ()Z
+		COMMENT {@return whether there is a passenger in control of this entity}
+		COMMENT
+		COMMENT @see #getPrimaryPassenger
+		COMMENT @see #getPassengersList
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getPassengerList
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 18]
```diff
@@ -533,76 +1001,242 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
 		ARG 7 yaw
 		ARG 8 pitch
 	METHOD method_5642 getPrimaryPassenger ()Lnet/minecraft/class_1297;
+		COMMENT {@return the passenger in control of this entity, or {@code null} if there is none}
+		COMMENT
+		COMMENT <p>Rideable entities should override this to return the entity. This is
+		COMMENT usually {@code #getFirstPassenger}.
+		COMMENT
+		COMMENT @see #hasPrimaryPassenger
+		COMMENT @see #getPassengersList
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getPassengerList
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 19]
```diff
@@ -830,54 +1800,183 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT {@return whether a player is riding this entity or any of its passengers}
+		COMMENT
+		COMMENT @implNote The default implementation is very inefficient.
+		COMMENT
+		COMMENT @see #getPassengerList
+		COMMENT @see #streamIntoPassengers
+		COMMENT @see #streamSelfAndPassengers
+		COMMENT @see #streamPassengersAndSelf
+		COMMENT @see #getPassengersDeep
+		COMMENT @see #hasPassengersDeep
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #hasPassengerDeep
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 20]
```diff
@@ -352,27 +569,118 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT
+		COMMENT @see #getFrozenTicks
+		COMMENT @see #setFrozenTicks
+		COMMENT @see #getFreezingScale
+		COMMENT @see #isFrozen
 	METHOD method_32316 canFreeze ()Z
+		COMMENT {@return whether the entity can freeze}
+		COMMENT
+		COMMENT @implNote Entities cannot be frozen if they are in the {@link
+		COMMENT EntityTypeTags#FREEZE_IMMUNE_ENTITY_TYPES} tag. In addition to this, {@link
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT net.minecraft.tag.EntityTypeTags#FREEZE_IMMUNE_ENTITY_TYPES} tag. In addition to this, {@link
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 21]
```diff
@@ -352,27 +569,118 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT LivingEntity} cannot be frozen if they are spectator or if they wear an
+		COMMENT item inside {@link ItemTag#FREEZE_IMMUNE_WEARABLES} tag.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT item inside {@link net.minecraft.tag.ItemTag#FREEZE_IMMUNE_WEARABLES} tag.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 22]
```diff
@@ -444,22 +812,57 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 	METHOD method_39759 hasCollidedSoftly (Lnet/minecraft/class_243;)Z
 		ARG 1 adjustedMovement
 	METHOD method_40071 shouldEscapePowderSnow ()Z
+		COMMENT {@return whether the entity should escape from powder snow}
+		COMMENT
+		COMMENT <p>This returns {@code true} if the entity is/was in powder snow and
+		COMMENT if it can freeze.
+		COMMENT
+		COMMENT @see #canFreeze
+		COMMENT @see #isFreezing
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #isFrozen
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 23]
```diff
@@ -352,27 +569,118 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see #setFrozenTicks
+		COMMENT @see #getFreezingScale
+		COMMENT @see #isFrozen
 	METHOD method_32316 canFreeze ()Z
+		COMMENT {@return whether the entity can freeze}
+		COMMENT
+		COMMENT @implNote Entities cannot be frozen if they are in the {@link
+		COMMENT net.minecraft.tag.EntityTypeTags#FREEZE_IMMUNE_ENTITY_TYPES} tag. In addition to this, {@link
+		COMMENT LivingEntity} cannot be frozen if they are spectator or if they wear an
+		COMMENT item inside {@link net.minecraft.tag.ItemTag#FREEZE_IMMUNE_WEARABLES} tag.
```
Comment (apple502j) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT item inside {@link net.minecraft.tag.ItemTags#FREEZE_IMMUNE_WEARABLES} tag.
```

**Comment by modmuss50** [comment]
Conflicts, sorry :) Once fixed im quite happy to merge this right away 👍

**Pull request merged**